std/traits.d

```diff
@@ -5364,16 +5364,24 @@ enum bool isNumeric(T) = is(NumericTypeOf!T) && !isAggregateType!T;
 /**
 Detect whether $(D T) is a scalar type (a built-in numeric, character or boolean type).
  */
-enum bool isScalarType(T) = isNumeric!T || isSomeChar!T || isBoolean!T;
+enum bool isScalarType(T) = is(T : real);
 
 ///
 @safe unittest
 {
     static assert(!isScalarType!void);
+    static assert( isScalarType!(immutable(bool)));
+    static assert( isScalarType!(immutable(byte)));
+    static assert( isScalarType!(immutable(short)));
     static assert( isScalarType!(immutable(int)));
     static assert( isScalarType!(shared(float)));
     static assert( isScalarType!(shared(const bool)));
+    static assert( isScalarType!(const(char)));
+    static assert( isScalarType!(const(wchar)));
     static assert( isScalarType!(const(dchar)));
+    static assert( isScalarType!(const(float)));
+    static assert( isScalarType!(const(double)));
+    static assert( isScalarType!(const(real)));
```

Comment:
I think there's no need to repeat `bool` and `float` with different qualifiers. Some tests should drop qualifiers to cover no qualifers. We're also missing `long`, and we could use unsigned integers instead of signed for some of them. I don't think we should bother to have every permutation, just a good covering of different combinations - especially as this is a documented unittest.